{
  String errorMsg="some message";
  Exception generic=new Exception(errorMsg);
  generic.printStackTrace();
  assertTrue(generic.getMessage().contains(errorMsg));
  ForeignException e=ForeignException.deserialize(ForeignException.serialize(srcName,generic));
  assertArrayEquals("Local stack trace got corrupted",generic.getStackTrace(),e.getCause().getStackTrace());
  e.printStackTrace();
  assertTrue(e.getCause().getCause() == null);
  assertTrue(e.getCause().getMessage().contains(errorMsg));
}
